ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -858,6 +861,8 @@ func (r *RayClusterReconciler) updateStatus(instance *rayiov1alpha1.RayCluster)
 	} else {
 		if utils.CheckAllPodsRunnning(runtimePods) {
 			instance.Status.State = rayiov1alpha1.Ready
+		} else {
+			instance.Status.State = rayiov1alpha1.Pending
```

Comment:
It's normal for a Ray cluster to gain pods that are in Pending state, particularly if autoscaling is enabled. With this change, the cluster could switch back and forth between Running and Pending. That seems undesirable. (The current reporting of state is not great either.)

Reply:
@DmitriGekhtman 
How do we differentiate between Pods are that are Pending due to autoscaler vs Pods being Pending during initial cluster setup?

Reply:
By the following funky logic:
The only cluster `State` transition in the current logic is `""` -> `Ready`.
If, after the initial cluster setup, we have a `Pending` pod due to scale change or failure, the `State` will not return to `""`.